src/dense.jl

```diff
@@ -48,12 +48,28 @@ end
 
 Base.length(dense::DenseOutput) = length(dense.opts.tout)
 
+tdir(ivp, solver::DenseOutput) = sign(solver.opts.tstop - ivp.t0)
+
 @compat function (::Type{DenseOutput{I}}){T,I}(ivp::IVP{T};
+                                               tstop = T(Inf),
+                                               tout::AbstractVector{T} = T[tstop],
                                                opts...)
+    if all(tout.>=ivp.t0)
+        tout = sort(tout)
```

Comment:
I don't understand you concern, maybe you need to explain more carefully.  My point was that `tout` needs to be sorted in `tdir` direction, if it's not then error.